### DISCUSSION TITLE
[resolved] Zasoby Centrum Wiadomości i nagłówki odpowiedzi

### DISCUSSION THREAD

**Discussion by tomasz-grzesik** [allegro/allegro-api#4802] [Suggestions - resolved]
Drodzy pogramiści Allegro,

prośba o sprawdzenie nagłówka content-encoding w odpowiedzi zasobów Centrum Wiadomości.
Czasem zdarza się, że jest gzip, ale nie można odkodować zawartości.

**Comment by AureliuszBrussy** [comment]
Czy możesz wskazać przykładowe trace-id takiej odpowiedzi?

**Reply by tomasz-grzesik** [reply]
Właśnie zwiększam szczegółowość logów. Jak będę wiedział coś jeszcze dam znać.

**Reply by tomasz-grzesik** [reply]
Przykładowy URL:
'https://api.allegro.pl/messaging/threads/XXxTvoihvS0E564jg6qHAdyj0tu4GNflCxlOvl8govt/messages'

**Reply by tomasz-grzesik** [reply]
Odpowiedź ma kod 500. Nagłówk odpowiedzi jeszcze szukam, ale możlie, że nie ma go w logach i tak łatwo się do niego niedobiorę.

**Reply by AureliuszBrussy** [reply]
Dobrze rozumiem, że w przypadku błędu 500 w nagłówku content-encoding zdarza się wartość gzip?

**Reply by tomasz-grzesik** [reply]
tak

**Reply by AureliuszBrussy** [reply]
Czy jesteś w stanie podać w formacie curl cały response - wraz ze wszystkim nagłówkami?

**Reply by tomasz-grzesik** [reply]
Tak, postaram się wydobyć te dane. Proszę jednak o trochę czasu.

**Reply by tomasz-grzesik** [reply]
Przesyłam nagłówki:
```
reply: 'HTTP/1.1 500 Internal Server Error\r\n'
header: content-encoding: gzip
header: expires: 0
header: cache-control: no-cache, no-store, max-age=0, must-revalidate
header: x-xss-protection: 1; mode=block
header: pragma: no-cache
header: x-frame-options: DENY
header: trace-id: 39172cb3bc2ff82c
header: date: Wed, 11 Aug 2021 11:33:51 GMT
header: strict-transport-security: max-age=31536000 ; includeSubDomains
header: x-content-type-options: nosniff
header: content-type: application/json; charset=utf-8
header: transfer-encoding: chunked
```

**Reply by AureliuszBrussy** [reply]
Dziękuję. Faktycznie przy błędzie 500 w odpowiedzi nie powinien się pojawić ten header. Wdrożymy odpowiednią poprawkę.

**Reply by tomasz-grzesik** [reply]
Informuję, że błąd nieustannie występuje:
"'Received response with content-encoding: gzip, but failed to decode it.', error('Error -3 while decompressing data: incorrect header check')"

**Reply by AureliuszBrussy** [reply]
Dziękuję za informację i przypomnienie tego tematu. Faktycznie, poprawka nie została jeszcze wdrożona. Wznowię ten temat i wrócę z informacjami.

**Reply by AureliuszBrussy** [reply]
@tomasz-grzesik wdrożyliśmy poprawkę. Problem nie powinien już wystąpić.

**Reply by tomasz-grzesik** [reply]
Dziękuję za pamięć, a zwłaszcza działanie.
